kroxylicious/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -36,17 +39,33 @@ public FilterChainFactory(List<FilterDefinition> filterDefinitions) {
      * @param filterDefinitions the candidates to validate.
      * @throws InvalidFilterConfigurationException If there are problems.
      */
-    public static void validateFilterDefinitions(Collection<FilterDefinition> filterDefinitions) {
+    private static void validateFilterDefinitions(PluginFactory<FilterFactory<?>> pluginFactory, Collection<FilterDefinition> filterDefinitions) {
         if (filterDefinitions == null || filterDefinitions.isEmpty()) {
             return;
         }
         var invalidFilters = filterDefinitions.stream()
-                .filter(Predicate.not(FilterDefinition::isDefinitionValid))
-                .map(FilterDefinition::type)
+                .map(fd -> {
+                    try {
+                        FilterFactory<?> filterFactory = pluginFactory.pluginInstance(fd.type());
+                        Class<?> configType = filterFactory.configType();
```

Comment:
This is the main place where the runtime uses `configType()`, and I don't think it really achieves much. So long as the `ServiceBasedPluginFactoryRegistry` does its job properly I don't _think_ the exception path gets taken. 

We could also export the `PluginFactoryRegistry` having a method to expose the discovered config type, which would work for all plugins, and not require the plugin interfaces to provide some method to return the config type.